backend/internal/gateway/gateway.go

```
@@ -173,36 +161,27 @@ func (g *Gateway) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+		body, bodyReadErr, bodyTooLarge = readBoundedRequestBody(r.Body)
+		if spec.Body == bodyJSON && bodyReadErr == nil && !bodyTooLarge {
+			if looksLikeJSON(body) {
+				parseErr = json.Unmarshal(body, &envelope)
+			} else if len(bytes.TrimSpace(body)) > 0 {
+				parseErr = errors.New("invalid json")
+			}
+			record.Streaming = envelope.Stream
+			recordSessionCapture(r, envelope)
+			g.captureModelSlug(r.Context(), requestID, strings.TrimSpace(envelope.Model))
+			if suppliedTraceID, ok := suppliedTraceID(r, envelope.LiteLLMMetadata.TraceID); ok && suppliedTraceID != traceID {
+				traceID = suppliedTraceID
```

golangci-lint (2.13.2)

[error] ineffectual assignment to traceID

(ineffassign)